### ISSUE TITLE
Bump HDF5 version?

### ISSUE THREAD

**Issue by sriharisundar** [NREL/PRAS#52]
Is it okay to bump HDF5 version? Seems like we are restricted to 0.16, but Sienna ecosystem requires > 0.17. Would it be a breaking change to update to 0.17?

Tests passed with this environment:
````
  [abce61dc] Decimals v0.4.1
  [31c24e10] Distributions v0.25.107
  [f67ccb44] HDF5 v0.17.1
  [62286e6e] MinCostFlows v0.1.2
  [a15396b6] OnlineStats v1.6.3
  [925886fa] OnlineStatsBase v1.6.3
  [74087812] Random123 v1.7.0
  [189a3867] Reexport v1.2.2
  [2913bbd2] StatsBase v0.34.2
  [f269a46b] TimeZones v1.13.0
  [ade2ca70] Dates
  [de0858da] Printf
  [9a3f8284] Random
````

**Comment by sriharisundar** [comment]
@GordStephen @scdhulipala

**Comment by GordStephen** [comment]
Closed in #56
